test/src/test/java/org/corfudb/runtime/view/stream/StreamAddressSpaceTest.java

```diff
@@ -40,7 +49,124 @@ public void testStreamAddressSpaceMerge() {
         assertThat(streamA.getTail()).isEqualTo(numStreamBEntries - 1);
 
         LongStream.range(streamBTrimMark + 1, numStreamBEntries).forEach(address ->
-                assertThat(streamA.getAddressMap().contains(address)).isTrue()
+                assertThat(streamA.contains(address)).isTrue()
         );
     }
+
+    @Test
+    public void constructorTest() {
+        StreamAddressSpace obj1 = new StreamAddressSpace(5L, Collections.emptySet());
+        assertThat(obj1.getTrimMark()).isEqualTo(5L);
+        assertThat(obj1.size()).isEqualTo(0L);
+
+        StreamAddressSpace obj2 = new StreamAddressSpace(5L, Collections.singleton(6L));
+        assertThat(obj2.getTrimMark()).isEqualTo(5L);
+        assertThat(obj2.size()).isEqualTo(1L);
+        assertThat(obj2.contains(6L)).isTrue();
+
+        StreamAddressSpace obj3 = new StreamAddressSpace(Collections.singleton(6L));
+        assertThat(obj3.getTrimMark()).isEqualTo(Address.NON_ADDRESS);
+        assertThat(obj3.size()).isEqualTo(1L);
+        assertThat(obj3.contains(6L)).isTrue();
+    }
+
+    @Test
+    @Ignore
+    public void testConstructorTrim() {
+        StreamAddressSpace obj1 = new StreamAddressSpace(2L, ImmutableSet.of(1L, 2L, 3L, 4L));
+        assertThat(obj1.size()).isEqualTo(2);
+        assertThat(obj1.contains(1L)).isFalse();
+        assertThat(obj1.contains(2L)).isFalse();
+        assertThat(obj1.contains(3L)).isTrue();
+        assertThat(obj1.contains(4L)).isTrue();
+    }
+
+    @Test
+    public void testTrim() {
+        StreamAddressSpace obj1 = new StreamAddressSpace(ImmutableSet.of(1L, 2L, 3L, 4L));
+        assertThat(obj1.size()).isEqualTo(4);
+        assertThat(obj1.contains(1L)).isTrue();
+        assertThat(obj1.contains(2L)).isTrue();
+        assertThat(obj1.contains(3L)).isTrue();
+        assertThat(obj1.contains(4L)).isTrue();
+
+        obj1.trim(2L);
+        assertThat(obj1.size()).isEqualTo(2);
+        assertThat(obj1.contains(1L)).isFalse();
+        assertThat(obj1.contains(2L)).isFalse();
+        assertThat(obj1.contains(3L)).isTrue();
+        assertThat(obj1.contains(4L)).isTrue();
+
+        obj1.trim(4L);
+        assertThat(obj1.size()).isEqualTo(0);
+        assertThat(obj1.contains(3L)).isFalse();
+        assertThat(obj1.contains(4L)).isFalse();
+
+    }
+
+    @Test
+    public void testCopy() {
+        StreamAddressSpace obj1 = new StreamAddressSpace();
+        StreamAddressSpace obj1Copy = obj1.copy();
+        assertThat(obj1Copy.getTrimMark()).isEqualTo(obj1.getTrimMark());
+        assertThat(obj1Copy.size()).isEqualTo(obj1.size());
```

Comment:
probably (for completeness) we can add an assertion to verify they're not the same reference..

Reply:
Done.